includes/class-amp-theme-support.php

```diff
@@ -2001,6 +2001,10 @@ public static function prepare_response( $response, $args = [] ) {
 
 		$dom = Document::fromHtml( $response );
 
+		if ( AMP_Validation_Manager::$is_validate_request ) {
+			AMP_Validation_Manager::remove_illegal_source_stack_comments( $dom );
```

Comment:
It makes sense that the source stack comments be removed for a validation request.